### DISCUSSION TITLE
Considered using BSL instead of AGPL?

### DISCUSSION THREAD

**Discussion by voxpelli** [renovatebot/renovate#11145] [Suggest an Idea]
Some things I see beneficial with [BSL (Business Source License)](https://mariadb.com/bsl-faq-adopting/) compared to AGPL:

1. Avoids vendor lockin by converting itself to a defined GPL-compatible opensource license at a set [Change Date (max 4 years)](https://mariadb.com/bsl-faq-adopting/#changedate) after release. That ensures that if Renovate itself was to go out of business or eg. stop investing in Renovate, then it would be possible to revive it down the line without the obstacle of an AGPL license.
2. It isn't as contagious and scary as AGPL. Many companies have a blanket ban on installing any AGPL licensed code within their networks, see eg. [Google's policy](https://opensource.google/docs/using/agpl-policy/)
3. It has been reviewed and improved in cooperation with Bruce Perens, original creator of the Open Source Definition, and when [reading his thoughts on it](https://perens.com/2017/02/14/bsl-1-1/) I really get the feeling that BSL is likely the most opensource-compatible/friendly commercial license out there.

I think the purpose of the AGPL-license for Renovate is to stay as transparent and opensource as possible, while still discouraging someone else from ripping it of and launching a competing service on top of it.

I think that goal can be achieved much better with BSL, as it's less scary, will over time convert back to a permissive license and in general be a more friendly alternative than AGPL.

Would be nice to overall see more projects move away from the AGPL/SSPL style of licenses and to more friendly licenses which causes less of lockins and less of a headache for corporate lawyers.

**Comment by rarkins** [comment]
I think a non-OSI license might deter more people than the current AGPL. WhiteSource Renovate On-Premises essentially makes it available under a EULA too, although only for GitHub and GitLab so far. 

Will respond in more detail next week when I'm not on mobile-only.

**Reply by voxpelli** [reply]
Good point 👍 

For reference, the BSL has a section around OSI [in it's FAQ](https://mariadb.com/bsl-faq-adopting/#osl). I kind of like that its up front about not being an OSI license in itself and thus in a way is more "honest" about its intent to block commercial use. That + that it defers to a GPL-compatible license after the change date.

In practice I think GPL-compatibility is more important for many than OSI-compliance.

**Reply by rarkins** [reply]
By the way, I assume you're aware that anyone fearing an AGPL license for a CLI tool doesn't understand the license? Ie the license for this tool isn't directly a problem, it's the companies with blanket policies towards GPL which is the problem

**Reply by voxpelli** [reply]
Absolutely, though I also can understand the appeal in such blanket policies towards AGPL (you do mean AGPL right?) as it becomes a very easy policy to follow and a hard one to accidentally break.

**Reply by rarkins** [reply]
Yes I do understand why they have that policy :)
